web/plugin-manager/src/components/pluginBlock.tsx

```diff
@@ -0,0 +1,183 @@
+import { SetStateAction, useMemo, useState } from "react";
+import { ArrowPathIcon, TrashIcon, PlayCircleIcon } from "@heroicons/react/24/outline";
+import axiosServices from "../services/axios";
+import { AxiosResponse } from "axios";
+import { Plugin, PluginProps, PluginStatus } from "../../../shared/interfaces/IPlugin";
+import { statusHelper } from "../helpers/statusHelpers";
+
+function PluginBlock(p: PluginProps) {
+    // Callback to set error on parent
+    function sendErrorData(errorType: string, errorMessage: string) {
+        p.setErrorData(errorType, errorMessage);
+    }
+    // Data to display
+    let dataMemoized = useMemo(() => {
+        console.log("p.props useMemo", p.props);
+        return p.props;
+    }, [p.props]);
+
+    // Toggle status state
+    const [toggleStatus, setStatus] = useState(statusHelper(p.props.status));
+
+    const [playStatusClassName, setPlayStatusClassName] = useState(playStatus());
+
+    // fetch info from plugin to get fresh data on demand
+    async function fetchPluginStatus(): Promise<AxiosResponse<string>> {
+        let result: AxiosResponse<string> = {} as AxiosResponse<string>;
+        try {
+            return (result = await axiosServices.getpluginInfo(dataMemoized.id));
+        } catch (error) {
+            sendErrorData(
+                "error",
+                `Plugins infos failed to get infos from plugin name : ${dataMemoized.name} on id: ${dataMemoized.id}`
+            );
+        }
+        return result;
+    }
+
+    // Launch or stop plugin
+    async function launchAndStopPlugins() {
+        let resultPluginInfo: AxiosResponse<string>;
+        // fetch info from plugin to get fresh data
+        resultPluginInfo = await fetchPluginStatus();
+        // Update data
+        setStatus(statusHelper(resultPluginInfo.data));
+
+        let result: AxiosResponse<number>;
+        if (!toggleStatus) {
```

Comment:
Why not handling the passing case first ?

Reply:
Because Start process first . it's opinionated.

Reply:
My practice to start with the positive case first is extarcted from [The art of readable code](https://mcusoft.files.wordpress.com/2015/04/the-art-of-readable-code.pdf), chapter 7 The Order of if/else Blocks